server/consumer.go

```diff
@@ -1922,8 +1922,8 @@ func (o *consumer) deleteNotActive() {
 				}
 				nca := js.consumerAssignment(acc, stream, name)
 				js.mu.RUnlock()
-				// Make sure this is not a new consumer with the same name.
-				if nca != nil && nca == ca {
+				// Make sure this is the same consumer assignment, and not a new consumer with the same name.
+				if nca != nil && reflect.DeepEqual(nca, ca) {
```

Comment:
De-flakes `TestJetStreamClusterGhostEphemeralsAfterRestart` which would flake with `Still have %d missing consumers`. This would happen because a server could come online and need to catchup on some consumer deletes. But the leader had made a snapshot, making this server process that snapshot, which overwrites the consumer assignment to have the exact same config and RAFT group, but using different pointers. That would leave these consumer assignments around, whereas they should have been deleted still.